api/grpcserver/v2alpha1/transaction_test.go

```diff
@@ -757,3 +757,40 @@ func TestToTxContents(t *testing.T) {
 		require.Equal(t, spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_VESTING_SPAWN, txType)
 	})
 }
+
+func TestEvictedTransaction(t *testing.T) {
+	req := require.New(t)
+
+	db := statesql.InMemoryTest(t)
+	ctrl := gomock.NewController(t)
+	syncer := NewMocktransactionSyncer(ctrl)
+	publisher := pubsubmocks.NewMockPublisher(ctrl)
+	txHandler := NewMocktransactionValidator(ctrl)
+	conState := NewMocktransactionConState(ctrl)
+
+	signer, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	tx := newTx(0, types.Address{}, signer)
+	require.NoError(t, db.WithTxImmediate(context.Background(), func(dtx sql.Transaction) error {
+		require.NoError(t, transactions.Add(dtx, &types.Transaction{
+			RawTx:    tx.RawTx,
+			TxHeader: nil,
+		}, time.Time{}))
+		return nil
+	}))
```

Comment:
No transaction needed:

```suggestion
	require.NoError(t, transactions.Add(db, &types.Transaction{
		RawTx:    tx.RawTx,
		TxHeader: nil,
	}, time.Now()))
```